### ISSUE TITLE
Firefox related bug - TypeError: 'addEventListener' called on an object that does not implement interface EventTarget.

### ISSUE THREAD

**Issue by nsisodiya** [webcomponents/webcomponentsjs#269]
I am playing with Custom Elements.
My code is working well in Chrome but its not working well in Firefox 36.

Somewhere in the code in am writing like this

``` js
// Caches a local reference to `Element.prototype` for faster access.
  var ElementProto = (typeof Element !== "undefined" && Element.prototype) || {};

  // Cross-browser event listener shims
  var elementAddEventListener = ElementProto.addEventListener || function(eventName, listener) {
    return this.attachEvent("on" + eventName, listener);
  };
elementAddEventListener.call(ele, eventName, callback, false);
```

ele is the customElement, whereas eventName is "click", callback is a normal function.

Same code is working well in Chrome.  I am using following version in bower.json

``` json
"webcomponents.js": "~0.6.0"
```

**Comment by addyosmani** [comment]
Could you share a complete test case on JSBin that reproduces the issue using 0.6.0?

**Comment by nsisodiya** [comment]
@addyosmani ,
Hi Solved the bug.
Actually, It was my mistake that I loaded my event handing library before webcomponent.js ! See the link -
http://jsbin.com/behoxohobo/3

This is working well in Chrome but failed in Firefox.
When I corrected the order of scripts , Things started working on Firefox too (as expected)!

I am still unable to get why this link is working well in Chrome. I think chrome is some smart checking algo for webcomponents.js.

My Observation.

In that small utility lib I am caching ```Element.prototype.addEventListener``` So my code was not working because I was using native addEventListener function on custom Element. But In chrome, the cached variable addEventListener was pointing to addEventListener function provided by webcomponents.js, so things were working in chrome.
